src/transformations.jl

```diff
@@ -223,84 +289,125 @@ julia> StatsBase.transform(dt, X)
  0.0  0.5  1.0
 ```
 """
-function fit(::Type{UnitRangeTransform}, X::AbstractArray{<:Real,2}; unit::Bool=true)
-    d, n = size(X)
-
-    tmin = X[:, 1]
-    tmax = X[:, 1]
-    for j = 2:n
-        @inbounds for i = 1:d
-            if X[i, j] < tmin[i]
-                tmin[i] = X[i, j]
-            elseif X[i, j] > tmax[i]
-                tmax[i] = X[i, j]
+function fit(::Type{UnitRangeTransform}, X::AbstractMatrix{<:Real};
+             dims::Union{Integer,Nothing}=nothing, unit::Bool=true)
+    if dims == 1
+        l, tmin, tmax = _extract_info(X)
+    elseif dims == 2
+        l, tmin, tmax = _extract_info(X')
+    elseif dims == nothing
+        Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
+        l, tmin, tmax = _extract_info(X')
+    else
+        throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
+    end
+
+    for i = 1:l
+        @inbounds tmax[i] = 1 / (tmax[i] - tmin[i])
+    end
+    return UnitRangeTransform(l, dims, unit, tmin, tmax)
+end
+
+function _extract_info(X::AbstractMatrix{<:Real})
+    n, l = size(X)
+    tmin = X[1, :]
+    tmax = X[1, :]
+    for j = 1:l
```

Comment:
This was enough, right?
```suggestion
    for j = 2:l
```

Reply:
This cause test failed. I think that the following `for i = 2:n` is enough.

Reply:
Ah sorry I didn't notice you had switched `i` and `j`.